### PR TITLE
Use DocumentBasedSearchClient for dynamic index discovery

### PR DESCRIPTION
**Pull request by entangled90**
## Description
In this PR a new method to retrieve index by alias has been added to DocumentBasedSearchClient.

Classes used by Optimize to detect "dynamic" indices have been migrated to using the new client and this new method.
Because these classes only need `DocumentBasedSearchClient` they can now be easily instantiated by all webapps, making it possible to perform backups from any webapp.

The backup integration will be done in a followup PR

## Related issues

relates  #25597

### CODE REVIEW

File: webapps-backup/src/main/java/io/camunda/webapps/backup/DynamicIndicesProvider.java
Comment (npepinpe) [inline webapps-backup/src/main/java/io/camunda/webapps/backup/DynamicIndicesProvider.java:12]: :+1:

File: search/search-client-opensearch/src/test/java/io/camunda/search/os/transformers/index/IndexAliasResponseTransformerTest.java
Comment (npepinpe) [inline search/search-client-opensearch/src/test/java/io/camunda/search/os/transformers/index/IndexAliasResponseTransformerTest.java:41]: :fearful: I mean it's test, so I guess whatever, but it's still funky looking :smile:

Comment (npepinpe) [reply search/search-client-opensearch/src/test/java/io/camunda/search/os/transformers/index/IndexAliasResponseTransformerTest.java:41]: Actually even then, it's a little hard to read. Can we extract just, say, `AliasDefinition` ? I think that will already make it more readable.

**Review by npepinpe — APPROVED** [review]
:+1: I would still like to have someone from Optimize give a stamp of approval, so if it can wait until tomorrow for that...otherwise, if it's blocking you, then we can go ahead and merge, but please give the Optimize team a heads up in their team channel about these changes.

**Review by RomanJRW — APPROVED** [review]
Looks reasonable to me 👍

File: optimize/upgrade/pom.xml
Comment (entangled90) [reply optimize/upgrade/pom.xml:31]: Yes, but I plan to wait for the `optimize-single-jar` to merge in main first as there will be probably conflicts in this files. I put temporarily `${project.version}` to have all these "search" modules at the same version to be able to build locally

**Comment by entangled90** [comment]
@RomanJRW  I added a build step in the last commit to build either `-pl optimize/backend -am` or `-pl optimize/upgrade -am` in the jobs as a way to build the dependencies while keeping intact the rest of the jobs.

File: .github/workflows/ci.yml
Comment (RomanJRW) [inline .github/workflows/ci.yml:627]:
```suggestion
          parameters: -f optimize/pom.xml test -PskipFrontendBuild -Dskip.docker
```

The profile already sets the `-Dskip.fe.build` to true, so you can remove that

File: .github/workflows/optimize-deploy-artifacts.yml
Comment (RomanJRW) [inline .github/workflows/optimize-deploy-artifacts.yml:58]:
```suggestion
        parameters: install -pl optimize/backend -am -Dquickly -PrunAssembly -T1
```

File: .github/workflows/ci.yml
Comment (RomanJRW) [inline .github/workflows/ci.yml:622]:
```suggestion
          parameters: -pl optimize/backend -am install -Dquickly
```
I think this can be simplified and have the same effect with just the `quickly` flag

File: .github/workflows/optimize-ci-build-reusable.yml
Comment (RomanJRW) [inline .github/workflows/optimize-ci-build-reusable.yml:87]:
```suggestion
          parameters: -pl optimize/backend -am install -Dquickly -P runAssembly -T 1
```
the quickly flag should also be enough here

File: .github/workflows/optimize-ci.yml
Comment (RomanJRW) [inline .github/workflows/optimize-ci.yml:105]:
```suggestion
          parameters: -pl optimize/backend -am install -Dquickly -P runAssembly -T 1
```

File: .github/workflows/optimize-ci.yml
Comment (RomanJRW) [inline .github/workflows/optimize-ci.yml:173]:
```suggestion
          parameters: -pl optimize/upgrade -am install -Dquickly -PrunAssembly -T1
```

File: .github/workflows/optimize-ci.yml
Comment (RomanJRW) [inline .github/workflows/optimize-ci.yml:398]:
```suggestion
          parameters: -pl optimize/upgrade -am install -Dquickly -PrunAssembly -T1
```

**Review by RomanJRW — APPROVED** [review]
Some simplifications possible, but generally LGTM

**Pull request merged**
